nmrfx-analyst-gui/src/main/java/org/nmrfx/processor/gui/FXMLController.java

```diff
@@ -1519,26 +1497,39 @@ protected ArrayList<Double> getBaselineRegions(String vecDimName) {
         return fracs;
     }
 
-    private void toggleSideBarAttributes(ToggleButton attributesButton, ToggleButton contentButton, ToggleButton processorButton) {
+    /**
+     * Switchs which NmrControlRightSideContent is displayed in the nmrControlRightSidePane.
+     * @param attributesButton The attributes toggle button.
+     * @param contentButton The content toggle button.
+     * @param processorButton The processor toggle button.
+     */
+    private void toggleNmrControlRightSideContent(ToggleButton attributesButton, ToggleButton contentButton, ToggleButton processorButton) {
         if (attributesButton.isSelected()) {
-            borderPane.setRight(attributesPane);
+            addAttributesContentToNmrControlRightSidePane(attributesController.getPane());
             attributesController.setAttributeControls();
-            attributesController.updateScrollSize(borderPane);
-            hideProcessorAction();
         } else if (contentButton.isSelected()) {
-            borderPane.setRight(contentPane);
+            addAttributesContentToNmrControlRightSidePane(contentController.getPane());
             contentController.update();
-            contentController.updateScrollSize(borderPane);
-            hideProcessorAction();
         } else if (processorButton.isSelected()) {
+            // separate method for adding to rightContentPane as processor controller has additional setup
             showProcessorAction();
-            borderPane.setRight(null);
         } else {
-            borderPane.setRight(null);
-            hideProcessorAction();
+            nmrControlRightSidePane.getChildren().clear();
         }
     }
 
+    /**
+     * Add the attributes or content pane to the nmrControlRightSidePane, as well as hiding any ProcessorControllers associated
+     * with the charts so that the nmrControlRightSidePane contents will not be replaced when the active chart changes.
+     * @param pane The attributes or content pane to add.
+     */
+    private void addAttributesContentToNmrControlRightSidePane(Pane pane) {
+        // Set the visibility of all processor charts to false so that content/attributes will not be closed when switching in multichart view
+        charts.stream().map(chart -> chart.getProcessorController(false)).filter(Objects::nonNull).forEach(ProcessorController::hide);
+        nmrControlRightSidePane.getChildren().clear();
+        nmrControlRightSidePane.getChildren().add(pane);
```

Comment:
Ah thats true, to make it clearer I switched to use BorderPane instead of StackPane and used setCentre (in the NmrControlRightSidePane) whenever possible, and avoid using getChildren